src/testdir/test_delete.vim

```diff
@@ -97,3 +97,48 @@ func Test_symlink_recursive_delete()
   call assert_equal(0, delete('Xdir4/Xfile'))
   call assert_equal(0, delete('Xdir4', 'd'))
 endfunc
+
+func Test_complicated_name_recursive_delete()
+  call mkdir('Xcomplicated')
+  call mkdir('Xcomplicated/[complicated-1 ]')
+  call mkdir('Xcomplicated/{complicated,2 }')
+  split Xcomplicated/Xfile
+  call setline(1, ['a', 'b'])
+  w
+  w Xcomplicated/\[complicated-1\ \]/Xfile
+  w Xcomplicated/\{complicated,2\ \}/Xfile
```

Comment:
These two lines don't work on Windows.
Maybe they should be:

```
w Xcomplicated/\[complicated-1\ ]/Xfile
w Xcomplicated/\{complicated,2\ }/Xfile
```

(Remove ```\``` before `]` and `}`.)